packages/components/src/tree-grid/README.md

```diff
@@ -0,0 +1,129 @@
+# TreeGrid
+
+## Table of contents
+
+1. [Development guidelines](#development-guidelines)
+2. [Related components](#related-components)
+
+## Development guidelines
+
+`TreeGrid`, `TreeGridRow`, and `TreeGridCell` are components used to create a tree hierarchy. They're not visually styled components, but instead help with adding keyboard navigation and roving tab index behaviours to tree grid structures.
+
+A tree grid is a hierarchical 2 dimensional UI component, for example it could be used to implement a file system browser.
+
+A tree grid allows the user to navigate using arrow keys. Up/down to navigate vertically across rows, and left/right to navigate horizontally between focusables in a row.
+
+For more information on a tree grid, see the following links:
+
+- https://www.w3.org/TR/wai-aria-practices/examples/treegrid/treegrid-1.html
+
+### Usage
+
+The `TreeGrid` renders both a `table` and `tbody` element, and is intended to be used with `TreeGridRow` (`tr`) and `TreeGridCell` (`td`) to build out a grid.
+
+```jsx
+function TreeMenu() {
+	return (
+		<TreeGrid>
+			<TreeGridRow level={ 1 } positionInSet={ 1 } setSize={ 2 }>
+				<TreeGridCell>
+					{ ( props ) => <Button onClick={ onSelect } { ...props }>Select</Button> }
```

Comment:
Does the element inside the cell have to be a button? I wouldn't expect that to be the typical case for this component. If we were to have just plain text, like in [the wai-aria example](https://www.w3.org/TR/wai-aria-practices/examples/treegrid/treegrid-1.html), where would focus be transferred to? (This is something we can iterate on later though)

Reply:
Yeah, that's a good point! I can think of a few ways to accomplish this, and it shouldn't be too difficult as `TreeGridCell` has all the information it needs.